node/src/service.rs

```diff
@@ -531,10 +535,13 @@ pub fn new_full<
 					};
 
 					let current_version = worker_handle.version();
+					// TODO: what does this check do?
```

Comment:
This check checks if the nonce we mined is still relevant - we might have a new block already